oneoffs/ladder_detector.py

```diff
@@ -0,0 +1,121 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+sys.path.insert(0, '.')
+
+import os
+
+from collections import Counter, defaultdict
+
+from absl import app
+from tqdm import tqdm
+from sgfmill import sgf, sgf_moves
+
+import oneoff_utils
+
+
+ADJACENT_FOR_LADDER = 20
+
+
+def distance(a, b):
```

Comment:
Distance is a scalar quantity. This should probably be called subtract or something